File: day1/hello/func.go
package main

import (
	"fmt"
	"os"
)

func main() {
	data, err := os.ReadFile("hello.txt")
	if err != nil {
		fmt.Println("파일 읽기 실패:", err)
		return
	}
	fmt.Println(string(data))
}

/*
func divide(a, b int) (quotient int, remainder int) {
	quotient = a / b
	remainder = a % b
	return // naked return — 자동으로 quotient, remainder 반환
}

func main() {
	quot, rem := divide(10, 3)
	fmt.Println(quot, rem) // 3 1
}
*/

/*
func divide(a, b int) (int, int) {
	return a / b, a % b
}

func main() {
	quot, rem := divide(10, 3)
	fmt.Println(quot, rem) // 3 1
}
*/

/*
// go
func add(a, b int) int {
	return a + b
}

func main() {
	fmt.Printf("%d\n", add(1, 2))
}
*/

File: day1/hello/func.c
#include <stdio.h>
#include <fcntl.h>
#include <errno.h>
#include <string.h>

#if 1

int main()
{
	int fd;
	fd = open("hello.txt", O_RDONLY);

    if( fd < 0 )
    {
       printf("strerror=%s\n", strerror(errno) );
       return -1;
    }
}
#endif


#if 0
int divide(int a, int b, int *remainder) {
    *remainder = a % b;
    return a / b;
}

int main()
{
    int remainder;
    int result;
    result = divide( 10, 3, &remainder);
    printf("result=%d, remainder=%d\n", result, remainder);
}
#endif

#if 0
// C
int add(int a, int b) {
    return a + b;
}

int main()
{
	printf("%d\n" , add(1,2));
	return 0;
}
#endif
